docs/tutorials/writing_docs.rst

```diff
@@ -0,0 +1,244 @@
+
+1. Very briefly summarize the document in the header.
+2. Outline the sub-topics by creating section headers.
+3. Write an introduction/overview that ties the sub-topics together.
+4. Write each sub-topic as if it is its own document, starting from step 1.
+
+   - When the ideas are too small for sub-sections, then it is time to
+     write paragraphs.
+   - If the ideas in a document require more than 4 levels of headings,
+     you may need to consider moving topics to their own documentation pages.
+   - In contrast, document length should not be a reason to break up a page.
+   - For very small ideas with little content, use a dictionary list.
+
+
+Headings
+````````
+Headings should tell the user precisely what the section is about, given
+the context of the heading structure. You'll have a table of contents
+to reinforce this hierarchy.
+
+- Bad example: "Writing the source". What source?
+- Better example: "Writing a result parser module". That module.
+- Best example: "Result parser module" within a "Plugin Files" section in a
+  tutorial document about writing result parser plugins. We
+  know by context this a plugin file and that we'll be writing it.
+
+Documentation Type
+``````````````````
+There are four types of documentation within Pavilion:
+
+Usage
+  The purpose of usage documentation is to give the reader a quick
+  introduction to the overall structure of Pavilion. It should not try to
+  completely document any subject, and should instead link to the **Feature**
+  documentation on that topic. It may be mildly conversational in tone. Most
+  directories have a "basics" document that is of this type.
+Tutorial
+  Tutorials are meant to guide the user through the process of using the
+  software or adding to it. They may be mildly conversational in tone. They
+  should be concise but *complete*. They should cover each step in the
+  process, why that step exists, what happens when things go right, and
+  how to resolve issues when things go wrong.
+Feature
+  Feature documentation should attempt to completely detail how to use each
+  feature of the software. It may, when necessary, cover limits or other
+  considerations imposed by implementation details. It should be as concise
+  as possible, and should not be conversational in tone.
+API
+  API documentation is primarily written as part of the code. It is organized
+  by hand in documents under ``docs/source``, but the content is generated
+  from doc-strings using Sphinx's autodoc feature. Every source file,
+  class, method and function should be documented here. Function/methods
+  should have their arguments, return values, and exceptions raised documented.
+  Public globals, class and instance variables should also be documented.
+
+Audience
+````````
+The audience for Pavilion and its documentation is a technical one. You can
+generally assume the your audience understands at least the following:
+
+- Basic computer science concepts like regular expressions.
```

Comment:
In order to ensure this, a set of 'prerequisite' links at the top level of our documentation to overview pages on these subjects might be nice.